File: Springboot Code/AgroRent/src/main/java/com/majorproj/agrorent/dto/EquipmentUpdateDto.java
package com.majorproj.agrorent.dto;

import org.springframework.web.multipart.MultipartFile;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class EquipmentUpdateDto {
	@NotBlank(message = "Name is required")
	private String name;
    private String description;
    private MultipartFile image;
    @Positive(message = "rentalprice must be positive")
    private double rentalPrice;
}

File: Dot Net Code/AgroRent/DTOs/EquipmentUpdateDto.cs
using System.ComponentModel.DataAnnotations;

namespace AgroRent.DTOs
{
    public class EquipmentUpdateDto
    {
        [StringLength(30)]
        public string? Name { get; set; }

        public string? Description { get; set; }

        public double? RentalPrice { get; set; }

        public bool? Available { get; set; }
    }
}
